src/ec2/model.ts

```diff
@@ -41,21 +42,39 @@ export class Ec2ConnectionManager {
         return new DefaultIamClient(this.regionCode)
     }
 
-    protected async getAttachedPolicies(instanceId: string): Promise<IAM.attachedPoliciesListType> {
+    protected async getAttachedPolicies(instanceId: string): Promise<IAM.AttachedPolicy[]> {
         const IamRole = await this.ec2Client.getAttachedIamRole(instanceId)
-        if (!IamRole) {
+        if (!IamRole?.Arn) {
             return []
         }
-        const iamResponse = await this.iamClient.listAttachedRolePolicies(IamRole!.Arn!)
-
-        return iamResponse.AttachedPolicies ?? []
+        try {
+            const attachedPolicies = await this.iamClient.listAttachedRolePolicies(IamRole.Arn)
+            return attachedPolicies
+        } catch (e) {
+            const errorMessage = `No policies attached to role: ${IamRole.Arn}.`
+            getLogger().error(`ec2: ${errorMessage}`)
+            throw ToolkitError.chain(e, errorMessage, { code: 'NoSuchEntity' })
```

Comment:
But we should also check the code of the `e` in this scope. We can't assume that is the only error returned by `listAttachedRolePolicies`. What is `e.code` when `listAttachedRolePolicies` fails for the case concerning this PR?

Reply:
that code is `NoSuchEntity`. I changed it so that the ToolkitError inherits the error code from the AWSError.